https://github.com/xbmc/xbmc/pull/25106

From 2bf5ab07731b3d4160196e6b8d9ab9e25bea2ef9 Mon Sep 17 00:00:00 2001
From: Rudi Heitbaum <rudi@heitbaum.com>
Date: Thu, 2 May 2024 08:02:42 +0000
Subject: [PATCH 1/2] add missing c++ headers

Add missing headers which are no longer indirectly included by
other headers, fixes build with gcc-14

Signed-off-by: Rudi Heitbaum <rudi@heitbaum.com>
--- a/xbmc/platform/posix/filesystem/SMBWSDiscoveryListener.cpp
+++ b/xbmc/platform/posix/filesystem/SMBWSDiscoveryListener.cpp
@@ -17,6 +17,7 @@
 
 #include "platform/posix/filesystem/SMBWSDiscovery.h"
 
+#include <algorithm>
 #include <array>
 #include <chrono>
 #include <mutex>

From b6ddd8fb25ef20e8adb9a9f9b53bebf0cee3a970 Mon Sep 17 00:00:00 2001
From: Rudi Heitbaum <rudi@heitbaum.com>
Date: Thu, 2 May 2024 08:05:00 +0000
Subject: [PATCH 2/2] Geometry: fix template-id not allowed for constructor

fixes gcc-14 error:
- warning: template-id not allowed for constructor in C++20

Signed-off-by: Rudi Heitbaum <rudi@heitbaum.com>
--- a/xbmc/utils/Geometry.h
+++ b/xbmc/utils/Geometry.h
@@ -177,7 +177,7 @@ template <typename T> class CSizeGen
     return {m_w, m_h};
   }
 
-  template<class U> explicit CSizeGen<T>(const CSizeGen<U>& rhs)
+  template<class U> explicit CSizeGen(const CSizeGen<U>& rhs)
   {
     CheckSet(static_cast<T> (rhs.m_w), static_cast<T> (rhs.m_h));
   }
